content/posts/2017/2017-10-12-tibbles.Rmd

```diff
@@ -0,0 +1,227 @@
+---
+title: "The Trouble with Tibbles"
+author: "Theo Roe"
+date: 2017-10-13
+output: html_document
+image: "img/2017/tibbles.png"
+draft: false
+---
+
+```{r setup,include=FALSE}
+knitr::opts_chunk$set(echo = TRUE)
+knitr::opts_chunk$set(collapse = TRUE)
+```
+
+Let's get something straight, there isn't really any trouble with tibbles. I'm hoping you've noticed this is a play on 1967 Star Trek episode, "The Trouble with Tribbles". I've recently got myself a job as a Data Scientist, here, at [Jumping Rivers](https://www.jumpingrivers.com/). Having never come across tibbles until this point, I now find myself using them in nearly every R script I compose. Be that your timeless standard R script, your friendly Shiny app or an analytical Markdown document. 
+
+## What are tibbles?
+
+Presumably this is why you came here, right?
+
+Tibbles are a modern take on data frames, but crucially they are _still_ data frames. Well, what's the difference then? There's a quote I found somewhere on the internet that decribes the difference quite well;
+
+ _"keeping what time has proven to be effective, and throwing out what is not"_.   
+
+Basically, some clever people took the classic `data.frame`, shook it til the ineffective parts fell out, then added some new, more appropriate features. 
+
+## Precursors
+
+```{r eval = FALSE}
+# The easiest way to get access is to isstall the tibble package.
+install.packages("tibble")
+
+# Alternatively, tibbles are a part of the tidyverse and hence installing the whole tidyverse will give you access.
+install.packages("tidverse")
+```
+```{r eval = TRUE}
+# I am just going to use tibble.
+library(tibble)
+```
+
+
+## Tribblemaking
+
+There are 3 ways to form a tibble. It pretty much acts as your friendly old pal `data.frame` does. Just like standard data frames, we can create tibbles, coerce objects into tibbles and import data sets into `R` as a tibble. Below is a table of the traditional `data.frame` commands and their respective `tidyverse` commands.
+    
+_Formation Type_| _Data Frame Commands_          | *Tibbles Commands*
+--------------- | -------------------------------| ---------------------------------------------
+_Creation_      | `data.frame`                   | `data_frame` `tibble` `tribble`
+_Coercion_      | `as.data.frame`                | `as_data_frame` `as_tibble` 
+_Importing_     | `read.*`                       | `read_delim` `read_csv` `read_csv2` `read_tsv`
+
+Let's take a closer look...
+
+#### 1) Creation. 
+Just as `data.frame` creates data frames,`tibble`, `data_frame` and `tribble` all create tibbles. 
+
+Standard data frame.
+```{r}
+data.frame(a = 1:5, 
+           b = letters[1:5])  
+```
+A tibble using `data_frame`.
```

Comment:
Probably dump the code for data_frame and just add a comment above the tibble code